model/src/main/kotlin/PackageCuration.kt

```diff
@@ -64,7 +70,24 @@ data class PackageCuration(
      */
     private fun isApplicableIvyVersion(pkgId: Identifier) =
         runCatching {
-            Semver.coerce(pkgId.version).satisfies(id.version)
+            if (Semver.isValid(id.version)) {
+                // If the curation is for a valid semver, do not coerce the package version, as it also has to be valid.
```

Comment:
Is this related to https://github.com/semver4j/semver4j/issues/130?

Reply:
It is related, but I actually added this check due to issue that 2.0.0 matched 2.0.0.3.
But investigating this more revealed even more issues.
We might need to remove the `coerce()` call for the version altogether: https://github.com/oss-review-toolkit/ort/pull/6430

Reply:
The more I think about all the issues we face, the more I come to believe we should create our own Kotlin multi-platform version comparison library...